src/php/ext/grpc/channel_credentials.c

```diff
@@ -116,8 +116,13 @@ PHP_METHOD(ChannelCredentials, setDefaultRootsPem) {
                          "setDefaultRootsPem expects 1 string", 1 TSRMLS_CC);
     return;
   }
-  default_pem_root_certs = gpr_malloc((pem_roots_length + 1) * sizeof(char));
-  memcpy(default_pem_root_certs, pem_roots, pem_roots_length + 1);
+  char *default_certs = php_grpc_get_default_pem_root_certs(TSRMLS_C);
+  if (default_certs == NULL) {
```

Comment:
This has been done in a different way now. The certs string is stored in the "persistent list", similarly to persistent channel. Everytime we call `setDefaultRootPem`, we check the persistent list. If it exists, we retrieve it and compare it to the incoming parameter. If they are different, I update the persistent list. 

Previously we just store the char pointer in a variable and apparently PHP will somehow corrupt the pointer once in a while so I cannot use a simple variable.